upstairs/src/dummy_downstairs_tests.rs

```diff
@@ -156,102 +156,123 @@ pub(crate) mod protocol_test {
         }
 
         pub async fn negotiate_start(&self) {
-            let packet = self
-                .fr
-                .lock()
-                .await
-                .next()
-                .await
-                .transpose()
-                .unwrap()
-                .unwrap();
-            match &packet {
-                Message::HereIAm {
-                    version,
-                    upstairs_id: _,
-                    session_id: _,
-                    gen: _,
-                    read_only: _,
-                    encrypted: _,
-                    alternate_versions: _,
-                } => {
-                    info!(self.inner.log, "negotiate packet {:?}", packet);
+            loop {
```

Comment:
currently, a ping can't happen before `HereIAm`, it only happens in the select loop after that packet has been sent.

Reply:
Done